projects/forms/src/lib/directives/errors/errors.directive.ts

```diff
@@ -0,0 +1,240 @@
+
+	/**
+	 * A reference to a control or a string reference to the control
+	 */
+	@Input('ngxErrors') public control: AbstractControl | string;
+
+	constructor(
+		@Optional() private readonly formGroupDirective: FormGroupDirective,
+		@Optional() private readonly formNameDirective: FormGroupName,
+		@Optional()
+		@Inject(NgxFormsErrorsConfigurationToken)
+		private readonly config: NgxFormsErrorConfigurationOptions,
+		private readonly viewContainer: ViewContainerRef,
+		private readonly elementRef: ElementRef,
+		private readonly renderer: Renderer2,
+		private readonly templateRef: TemplateRef<any>
+	) {
+		// Iben: Set the current template ref at constructor time so we actually have the provided template (as done in the *ngIf directive)
+		this.template = this.templateRef;
+	}
+
+	public ngOnDestroy(): void {
+		// Iben: Handle the on destroy flow
+		this.onDestroySubject$.next();
+		this.onDestroySubject$.complete();
+	}
+
+	public ngAfterViewInit(): void {
+		// Iben: Render the actual input so that it is always visible
+		this.viewContainer.clear();
+		this.viewContainer.createEmbeddedView(this.template);
+
+		// Iben: If no control was provided, we early exit and log an error
+		if (!this.control) {
+			console.error('No control was provided to the NgxFormsErrorDirective');
+
+			return;
+		}
+
+		// Iben: If the control is a string, we check the parent to find the actual control.
+		// If not, we use the provided control
+		if (typeof this.control === 'string') {
+			this.abstractControl = this.formGroupDirective
+				? this.formGroupDirective.form.get(this.control)
+				: this.formNameDirective?.control.get(this.control);
+		} else {
+			this.abstractControl = this.control;
+		}
+
+		// Iben: If no control was found, we early exit and log an error
+		if (!this.abstractControl) {
+			console.error('No control was provided to the NgxFormsErrorDirective');
+
+			return;
+		}
+
+		// Iben: Listen to the value changes and the touched changes of the control
+		combineLatest([
+			this.abstractControl.valueChanges.pipe(startWith(this.abstractControl.value)),
+			touchedEventListener(this.abstractControl),
+		])
+			.pipe(
+				tap(([, touched]) => {
+					// Iben: Check whether we should show the error based on the provided config
+					const shouldShow =
+						this.abstractControl.invalid &&
+						(this.config.showWhen === 'touched' ? touched : this.abstractControl.dirty);
+
+					// Iben: Show the error based on whether or not a component was provided
+					if (!this.config.component) {
+						this.handleNoComponentFlow(shouldShow);
+					} else {
+						this.handleComponentRender(shouldShow);
+					}
+				}),
+				takeUntil(this.onDestroy$)
+			)
+			.subscribe();
+	}
+
+	/**
+	 * Renders a provided custom component underneath the input component
+	 *
+	 * @param shouldShow - Whether the error should be shown
+	 */
+	private handleComponentRender(shouldShow: boolean) {
```

Comment:
Ik zou hier een default value voorzien voor `shouldShow` zodat die niet breekt als er for some reason geen value meegegeven wordt. Idem hieronder.

Reply:
Welke meerwaarde heeft dat? Puur uit nieuwsgierigheid, we willen het enkel tonen als het expliciet true is, dus als het null of undefined zou zijn dan gaat het nog steeds niet getoond worden.

Reply:
Hm fair, je doet inderdaad een falsy check op die value. Gohja enige meerwaarde dan zou zijn dat je impliciet terug valt op een false ipv. falsy value. In dit geval misschien niet echt de moeite.